### ISSUE TITLE
Classifier: Can't use keyboard to dismiss transcription subtask when text field is empty

### ISSUE THREAD

**Issue by goplayoutside3** [zooniverse/front-end-monorepo#6828]
## Package
- lib-classifier

## Describe the bug
The transcription subtask (component called SubTaskPopup) asks for confirmation (ConfirmModal) when trying to close the subtask modal with an empty text field. The ConfirmModal is inaccessible from the keyboard. This is a pattern of two modals rendering at the same time.

## To Reproduce
1. Go to https://www.zooniverse.org/projects/blicksam/transcription-task-testing/classify
2. Click on an unfinished transcribed line or create a new one
3. Close the subtask modal via the Esc key
4. The confirm modal appears and cannot be accessed via the keyboard

## Screenshots
![Image](https://github.com/user-attachments/assets/5e9edc11-dbd4-4715-816f-c609e29cb0c0)

**Comment by goplayoutside3** [comment]
This is a duplicate of https://github.com/zooniverse/front-end-monorepo/issues/3314, though the suggested solution in that Issue is we shouldn't expect a keyboard user to have to use a mouse or trackpad to confirm a decision.

**Comment by eatyourgreens** [comment]
My suggestion in #3314 was to allow `Esc` to close the second dialog. Grommet also has some focus bugs that are triggered when you open a `Layer` on top of a `Layer`.

**Comment by eatyourgreens** [comment]
Converting those popups to use `<dialog>`, instead of `Layer`, might fix this.

**Comment by eatyourgreens** [comment]
Thank you for opening this, by the way. 🙏🏻

**Comment by eatyourgreens** [comment]
Another option might be to use [`confirm()`](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm), in which case the browser will handle the keyboard for you. `Enter` to confirm, `Esc` to cancel, I think.

**Comment by eatyourgreens** [comment]
The confirmation modal should be marked up as [an alert dialog](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/alertdialog_role) too.

**Comment by eatyourgreens** [comment]
The [Alert and Message Dialog Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/) has examples of keyboard shortcuts, and roles and states, for a modal confirmation dialog.
